### DISCUSSION TITLE
How to define a custom animation / keyframes with Twind?

### DISCUSSION THREAD

**Discussion by zaynetro** [freshframework/fresh#1542] [Q&A] [answered]
Tailwind CSS allows to defined custom animation and keyframe values in user config: https://tailwindcss.com/docs/animation#using-custom-values

How can I achieve something similar with Twind?

My current solution is to manually include keyframe styles on the page:

```ts
const customStyles = `
@keyframes fadeIn {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
`;

...

<style dangerouslySetInnerHTML={{ __html: customStyles }} />
```

and then 

```html
<div class="animate-[fadeIn_0.2s_ease-in-out]">
```

**Comment by marvinhagemeister** [comment]
See the twind docs: https://twind.dev/handbook/css-in-twind.html#the-keyframes-function

**Reply by zaynetro** [reply, marked as answer]
Thanks! I was reading their new https://twind.style/ website and couldn't find anything I could understand about animations. Luckily, the examples from the old website worked.

So fade in animation using twind v1 could be done like this

```ts
import { animation, keyframes } from "@twind/core";

const fadeIn = animation(
  "0.2s ease-in-out",
  keyframes({
    "0%": {
      opacity: 0,
    },
    "100%": {
      opacity: 1,
    },
  }),
);
```

```tsx
<div class={`${fadeIn}`}>
```
